test/Test.cpp

```diff
@@ -120,41 +164,51 @@ int main(int,char**)
 	java::lang::Object object = java::lang::Integer(23754);
 	if (jni::InstanceOf<java::lang::Number>(object) && jni::Cast<java::lang::Number>(object))
 	{
-		printf("%d\n", static_cast<int>(java::lang::Number(object)));
+		printf("%d\n", static_cast<jint>(java::lang::Number(object)));
 	}
 	else
 	{
-		int* p = 0; *p = 3;
+		exit(-1);
```

Comment:
with that *p = 3 hack, the failure would look like this
```
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x000000010f6a0837, pid=2125, tid=0x0000000000000103
#
# JRE version: OpenJDK Runtime Environment (8.0) (build 1.8.0-adoptopenjdk-jenkins_2018_05_19_02_01-b00)
# Java VM: OpenJDK 64-Bit Server VM (25.71-b00 mixed mode bsd-amd64 compressed oops)
# Problematic frame:
# C  [JNIBridgeTests+0x1837]  main+0x877
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /Users/tomasdirvanauskas/Projects/android-jni-bridge/hs_err_pid2125.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#
Test failed at ./build.pl line 289, <CONFIG> line 2.
```

with exit (-1); 

you would get only this
`
Test failed at ./build.pl line 289, <CONFIG> line 2.`

which is much cleaner, but I see it's lacking info, i'll switch to assert instead, since provide line numbers